core/trino-main/src/main/java/io/trino/sql/planner/plan/TableWriterNode.java

```diff
@@ -530,24 +599,34 @@ public String toString()
         {
             return handle.map(Object::toString).orElse("[]");
         }
+
+        @Override
+        public boolean supportsReportingWrittenBytes(Metadata metadata, Session session)
+        {
+            throw new UnsupportedOperationException();
+        }
     }
 
     public static class TableExecuteTarget
             extends WriterTarget
     {
         private final TableExecuteHandle executeHandle;
         private final Optional<TableHandle> sourceHandle;
+        private final TableHandle tableHandle;
```

Comment:
As discussed offline:
* lets compute the value already in `io.trino.sql.planner.LogicalPlanner#createTableExecutePlan` and store boolean in `TET`
* then in `supportsReportingWrittenBytes` compute based on `sourceHandle` and if not set return precomputed flag.

Reply:
@losipiuk, done